game-core/src/test/java/games/strategy/triplea/delegate/MoveValidatorTest.java

```diff
@@ -117,45 +117,45 @@ void testValidateMoveForRequiresUnitsToMove() throws Exception {
     List<Unit> toMove = berlin.getUnitCollection().getMatches(Matches.unitCanMove());
     MoveValidationResult results =
         MoveValidator.validateMove(
-            toMove, r, germans, Collections.emptyList(), new HashMap<>(), false, null, twwGameData);
+            toMove, r, germans, Map.of(), Map.of(), false, null, twwGameData);
     assertTrue(results.isMoveValid());
 
     // Add germanTrain to units which fails since it requires germainRail
     addTo(berlin, GameDataTestUtil.germanTrain(twwGameData).create(1, germans));
     toMove = berlin.getUnitCollection().getMatches(Matches.unitCanMove());
     results =
         MoveValidator.validateMove(
-            toMove, r, germans, Collections.emptyList(), new HashMap<>(), false, null, twwGameData);
+            toMove, r, germans, Map.of(), Map.of(), false, null, twwGameData);
     assertFalse(results.isMoveValid());
 
     // Add germanRail to only destination so it fails
     final Collection<Unit> germanRail = GameDataTestUtil.germanRail(twwGameData).create(1, germans);
     addTo(easternGermany, germanRail);
     results =
         MoveValidator.validateMove(
-            toMove, r, germans, Collections.emptyList(), new HashMap<>(), false, null, twwGameData);
+            toMove, r, germans, Map.of(), Map.of(), false, null, twwGameData);
     assertFalse(results.isMoveValid());
 
     // Add germanRail to start so move succeeds
     addTo(berlin, GameDataTestUtil.germanRail(twwGameData).create(1, germans));
     results =
         MoveValidator.validateMove(
-            toMove, r, germans, Collections.emptyList(), new HashMap<>(), false, null, twwGameData);
+            toMove, r, germans, Map.of(), Map.of(), false, null, twwGameData);
     assertTrue(results.isMoveValid());
 
     // Remove germanRail from destination so move fails
     GameDataTestUtil.removeFrom(easternGermany, germanRail);
     results =
         MoveValidator.validateMove(
-            toMove, r, germans, Collections.emptyList(), new HashMap<>(), false, null, twwGameData);
+            toMove, r, germans, Map.of(), Map.of(), false, null, twwGameData);
```

Comment:
We should probably confirm this with a convention, though the `Map.of()` seems a bit odd when there is `Collections.emptyMap()`. I feel like the var-args variant should be used when there is at least one parameter. For example, it would also be a bit odd and seemingly indirect to use `Map.of(new Unit[0[)` (which is equivalent to `Map.of()`), when `Collections.emptyMap()` is the same thing and more direct (and there is no unused parameter that is passed)

Reply:
I agree that it would be good to make a decision on this and then apply it consistently going forward.

Here's why I think Map.of() is better than Collections.emptyMap() and similar for List.of() vs Collections.emptyList():

- It's shorter to write. This makes a difference for code readability especially when used in long parameter lists - often allowing the code to be formatted much more concisely.
- Map.of() is already useful for cases where the size is non-0. If we'll already be using it for those cases, it's better to remember just one API rather than multiple - i.e. no need to ever use Collections.emptyList().

Let me know if this makes sense - if so, I suggest documenting this somewhere - and maybe as a follow up, we can do a bulk replace of existing Collections.emptyList() and emptyMap() uses to make everything consistent.

Reply:
@asvitkine this is a matter of what one is used to, which is okay, we can set expectations.

I'm curious, in your opinion, doesn't `Map.of()` seem a bit indirect though? It's another way of saying "emptyMap", hence the indirection. In other words, instead of saying "empty map", we're saying "map with zero elements", it's the same, but just slightly more round-about IMO. Is the analogy to "size() == 0" vs "empty()" a reasonable analogy, or do you view that differently?

Either way, we don't have a solid convention one way or another. Consistency argument would suggest to use 'empty*", but I don't think it's worthwhile to hold things up. We can extract this line of discussion to: https://github.com/triplea-game/triplea/issues/5489